core/src/main/kotlin/io/github/bluegroundltd/outbox/TransactionalOutboxBuilder.kt

```diff
@@ -163,6 +164,20 @@ class TransactionalOutboxBuilder(
     return this
   }
 
+  /**
+   * Flag which indicates whether new instant processing mechanism should be used or not.
+   *
+   * The new mechanism uses monitor to fetch the item along with its group sibling items and support ordering.
+   * The old mechanism processes the instant outbox item directly without fetching it from store.
+   *
+   * - true:  Use the new mechanism for instant processing and allow ordering of grouped items.
+   * - false: Use the old mechanism for instant processing.
+   */
```

Comment:
Given that going forward we would like to have this enabled by default, perhaps it makes sense to add something about the default values.

```suggestion
  /**
   * Flag which indicates whether new instant processing mechanism should be used or not.
   *
   * The new mechanism uses monitor to fetch the item along with its group sibling items and support ordering.
   * The old mechanism processes the instant outbox item directly without fetching it from store.
   *
   * - true:  Use the new mechanism for instant processing and allow ordering of grouped items.
   * - false: Use the old mechanism for instant processing.
   *
   * If not explicitly set, the value defaults to `false`. This minimizes unexpected disruption to applications
   * already using the transactional outbox. However, in the coming versions this will change and the value
   * will default to `true`. So, any applications that wish to have this feature disabled, should explicitly set it.
   */
```